lib/bundler/process_lock.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Bundler
+  class ProcessLock
+    def self.lock(bundle_path = Bundler.bundle_path)
+      lock_file_path = File.join(bundle_path, "bundler.lock")
+
+      File.open(lock_file_path, "w") do |f|
+        f.flock(File::LOCK_EX)
+
+        yield
+      end
+    rescue Errno::ENOLCK # NFS
```

Comment:
Good point, something I had not considered, I wonder if we want to actually handle it or if we just ignore using a lock in this scenario.